module/Makefile.in

```diff
@@ -27,25 +27,38 @@ export ZFS_MODULE_CFLAGS ZFS_MODULE_CPPFLAGS
 
 SUBDIR_TARGETS = icp lua
 
-modules:
+all: modules
+distclean maintainer-clean: clean
+install: modules_install
+uninstall: modules_uninstall
+check:
+
+modules-Linux:
 	list='$(SUBDIR_TARGETS)'; for targetdir in $$list; do \
 		$(MAKE) -C $$targetdir; \
 	done
-	$(MAKE) -C @LINUX_OBJ@ M=`pwd` @KERNEL_MAKE@ CONFIG_ZFS=m $@
+	$(MAKE) -C @LINUX_OBJ@ M=`pwd` @KERNEL_MAKE@ CONFIG_ZFS=m modules
+
+modules-unknown:
+	@true
+
+modules: modules-@ac_system@
 
-clean:
+clean-Linux:
 	@# Only cleanup the kernel build directories when CONFIG_KERNEL
 	@# is defined.  This indicates that kernel modules should be built.
-@CONFIG_KERNEL_TRUE@	$(MAKE) -C @LINUX_OBJ@ M=`pwd` @KERNEL_MAKE@ $@
+@CONFIG_KERNEL_TRUE@	$(MAKE) -C @LINUX_OBJ@ M=`pwd` @KERNEL_MAKE@ clean
 
 	if [ -f @LINUX_SYMBOLS@ ]; then $(RM) @LINUX_SYMBOLS@; fi
 	if [ -f Module.markers ]; then $(RM) Module.markers; fi
 
 	find . -name '*.ur-safe' -type f -print | xargs $(RM)
 
-modules_install:
+clean: clean-@ac_system@
+
+modules_install-Linux:
 	@# Install the kernel modules
-	$(MAKE) -C @LINUX_OBJ@ M=`pwd` $@ \
+	$(MAKE) -C @LINUX_OBJ@ M=`pwd` modules_install \
```

Comment:
The updated install target results in the `spl.ko` file being install in a slightly different location.  This works in my initial testing, but could cause problems for end users or any scripts expecting to find them is the previous location.  Let's tweak this to preserve the original location.

```
./extra/avl/zavl.ko
./extra/icp/icp.ko
./extra/lua/zlua.ko
./extra/nvpair/znvpair.ko
./extra/os/linux/spl/spl.ko (used to be installed as /extra/spl/spl.ko)
./extra/unicode/zunicode.ko
./extra/zcommon/zcommon.ko
./extra/zfs/zfs.ko
```

Reply:
It's just basing it off the source path. How do I special case one directory?

Reply:
I don't see any way to do that without either changing everything to point to `module/spl`for objects but keep sources in `module/os/linux/spl` - which would be super messy. I'd rather not unless this is known to actually break something.

Reply:
Never mind. I changed the build to build spl.ko in `module/spl`. :man_shrugging:

Reply:
My specific concern was the changing the location could lead to problems when upgrading/downgrading the packages.   Thanks, for fixing this!